graylog2-server/src/main/java/org/graylog2/database/pagination/MongoPaginationHelper.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.database.pagination;
+
+import com.mongodb.client.model.Collation;
+import org.bson.conversions.Bson;
+import org.graylog2.database.MongoEntity;
+import org.graylog2.database.PaginatedList;
+
+import java.util.function.Predicate;
+
+public interface MongoPaginationHelper<T extends MongoEntity> {
+    /**
+     * Sets the query filter to apply to the query.
+     *
+     * @param filter the filter, which may be null.
+     * @return A new pagination helper with the setting applied
+     */
+    MongoPaginationHelper<T> filter(Bson filter);
+
+    /**
+     * Sets the sort criteria to apply to the query.
+     *
+     * @param sort the sort criteria, which may be null.
+     * @return A new pagination helper with the setting applied
+     */
+    MongoPaginationHelper<T> sort(Bson sort);
+
+    /**
+     * Sets the sort criteria to apply to the query.
+     *
+     * @param fieldName the name of the field to sort by.
+     * @param order     "desc" to request descending sort order. Otherwise, by default, ascending order is used.
+     * @return A new pagination helper with the setting applied
+     */
+    MongoPaginationHelper<T> sort(String fieldName, String order);
```

Comment:
I wonder if we want to use an enum instead of a string for the `order` parameter. :thinking: A string can be error-prone.

Reply:
This method is just the equivalent of https://github.com/Graylog2/graylog2-server/blob/81e34a5b17d91c9f8428a83bce7612eedd3e2a26/graylog2-server/src/main/java/org/graylog2/database/PaginatedDbService.java#L269

I think we usually pass the `order` parameter as a String right through from the REST resource.

Should we introduce an Enum type that we use as QueryParam in the REST resources instead? We could add a convenience method to make it easy for users to create the appropriate Bson structure, like `Sorts.descencing(...)` from that type. Then, we could probably even remove this method.

Reply:
I went ahead and implemented this to see how it feels. Let me know if you think it's worthwile to keep it. Otherwise we can revert the change.